kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/filter/EventLoopFilterThreadExecutor.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.filter;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.netty.channel.EventLoop;
+
+import io.kroxylicious.proxy.filter.FilterThreadExecutor;
+
+public class EventLoopFilterThreadExecutor implements FilterThreadExecutor {
+
+    private final EventLoop eventLoop;
+
+    public EventLoopFilterThreadExecutor(EventLoop eventLoop) {
+        this.eventLoop = eventLoop;
+    }
+
+    @Override
+    public boolean isInFilterThread() {
+        return eventLoop.inEventLoop();
+    }
+
+    @Override
+    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+        return eventLoop.schedule(command, delay, unit);
```

Comment:
Should this (and every other submit method) chain the result through `completedOnFilterThread`?